doc/src/modules/matrices/random.rst

```diff
@@ -0,0 +1,222 @@
+.. _matrices-random:
+
+Generate Specific Random Matrices
+=================================
+
+In order to setup distinct mathematical problems, like
+
+ - systems of linear equations,
+ - invertible matrices,
+ - triangular matrices
+ - differential equations with given eigenvalues
+ - linear isometries
+ - non-standard geometries by non-standard scalar products
+ - orthogonal basis of vector space,
+
+specific types of matrices are required.
+
+A purely random generation of matrix entries does not grant any
+specific shape or property of the matrix.
+So, those matrices have to be created carefully using
+classical classification theorems of linear algebra.
+
+Generators on different types of matrices are presented below.
+
+.. note::
+
+   Even for carefully sampled matrices which are elements in topological matrix groups,
+   the sample distribution does not claim to meet any uniform distribution
+   (with respect to the groups Haar measure).
+   So the sampling procedures may be used to produce educational or academic problems
+   but do not generate proper pre-defined distribution.
```

Comment:
Also worth mentioning that the matrices are useful for testing algorithms. I expect we can build some good randomised testing from these that would show up bugs in many of the matrix routines.